plugins/parquet/include/parquet/operator.hpp

```diff
@@ -115,6 +116,72 @@ class parquet_parser final : public plugin_parser {
   }
 };
 
+auto remove_empty_records(std::shared_ptr<arrow::Schema> schema)
+  -> std::shared_ptr<arrow::Schema> {
+  auto impl = [](const auto& impl, std::shared_ptr<arrow::DataType> type)
+    -> std::shared_ptr<arrow::DataType> {
+    TENZIR_ASSERT(type);
+    if (const auto* list_type = try_as<arrow::ListType>(type.get())) {
+      return arrow::list(impl(impl, list_type->value_type()));
+    }
+    if (const auto* struct_type = try_as<arrow::StructType>(type.get())) {
+      if (struct_type->num_fields() == 0) {
```

Comment:
I think we want a warning here, given that we are throwing away data. I am not sure how much effort we want to put into making it properly print the (nested) fieldname.

Reply:
Found a good way to get this done:

```
❯ tenzir --tql2 'from {x: [[{y: {}}]]} | write_parquet | read_parquet'
warning: replacing empty record with null at `x[][].y`
 = note: empty records are not supported in Apache Parquet
{
  "x": [
    [
      {
        "y": null
      }
    ]
  ]
}
```